kubernetes.go

```diff
@@ -82,11 +85,15 @@ func NewK8sPool(conf K8sPoolConfig) (*K8sPool, error) {
 		return nil, errors.Wrap(err, "during NewForConfig()")
 	}
 
+	ctx, fn := context.WithCancel(context.Background())
+
 	pool := &K8sPool{
-		done:   make(chan struct{}),
-		log:    conf.Logger,
-		client: client,
-		conf:   conf,
+		done:        make(chan struct{}),
+		log:         conf.Logger,
+		client:      client,
+		conf:        conf,
+		watchCtx:    ctx,
+		cancelWatch: fn,
```

Comment:
To be consistent, I'd like to see `watch` as the prefix for both these new fields.